### ISSUE TITLE
[BUG] Prometheus is not able to scrape metrics from AKS/EKS nodes.

### ISSUE THREAD

**Issue by przemyslavic** [epiphany-platform/epiphany#1884]
**Describe the bug**
Prometheus is not able to scrape metrics from AKS/EKS nodes .

**How to reproduce**
1. Deploy the AKS / EKS cluster with monitoring enabled.

**Expected behavior**
Prometheus configuration allows you to scrape metrics from node exporters installed as DaemonSets.

**Config files**
You can use the `m-generic-epiphany-on-aks` module with the following config:
```yaml
kind: azepi-config
azepi:
  vault_password: "test"
  config: |
    kind: epiphany-cluster
    title: Epiphany cluster Config
    name: azepi
    provider: any
    specification:
      name: azepi
      admin_user:
        name: operations
        key_path: ~/vms_rsa
      cloud:
        k8s_as_cloud_service: true
      components:
        repository:
          count: 1
          machines:
            - default-qa-monitoring-tests-all-0
        kubernetes_master:
          count: 0
        kubernetes_node:
          count: 0
        logging:
          count: 1
          machines:
            - default-qa-monitoring-tests-all-1
        monitoring:
          count: 1
          machines:
            - default-qa-monitoring-tests-all-0
        kafka:
          count: 0
        postgresql:
          count: 0
        load_balancer:
          count: 0
        rabbitmq:
          count: 0
    ---
    kind: configuration/feature-mapping
    title: Feature mapping to roles
    name: azepi
    provider: any
    specification:
      roles_mapping:
        repository:
          - repository
          - image-registry
          - firewall
          - filebeat
          - node-exporter
          - applications
    ---
    kind: infrastructure/machine
    name: default-qa-monitoring-tests-all-0
    provider: any
    specification:
      hostname: qa-monitoring-tests-all-0
      ip: x.x.x.x
    ---
    kind: infrastructure/machine
    name: default-qa-monitoring-tests-all-1
    provider: any
    specification:
      hostname: qa-monitoring-tests-all-1
      ip: y.y.y.y
```

**Environment**
- Cloud provider: [all]
- OS: [all]

**Additional context**
There are missing entries for node exporters in `k8s as cloud service` in the Prometheus configuration file found on the monitoring machine:
```
root@qa-monitoring-tests-all-0:~# cat /etc/prometheus/prometheus.yml
# Ansible managed
# http://prometheus.io/docs/operating/configuration/

global:
  scrape_interval: 15s #15s
  scrape_timeout: 10s #10s
  evaluation_interval: 10s
  external_labels:
    environment: qa-monitoring-tests-all-0




rule_files:
  - /etc/prometheus/rules/*.rules


scrape_configs:
  - job_name: "prometheus"
    metrics_path: "/metrics"
    static_configs:
      - targets:
          - "qa-monitoring-tests-all-0:9090"
  - job_name: "file_service_discovery"
    file_sd_configs:
      - files:
          - "/etc/prometheus/file_sd/*.yml"
```
The entire block of code responsible for adding scrape config for node exporters in `k8s as cloud service` is not executed, because the condition `{% if bearer_token is defined %}` that can be found in the template is not met.
The bearer token is set as fact in the task from the file `configure-k8s-apps-monitoring.yml`, which is run only when the condition `does_k8s_exists` is met, and this in turn is evaluated with the following condition `does_k8s_exists: "(groups ['kubernetes_master'] | default ( [])) | length> 0 "`, which is never true for AKS / EKS (no kubernetes_master group).

In short, we do not have a configuration for Prometheus to retrieve metrics from AKS / EKS nodes.
By the way the variable name `does_k8s_exists` can be changed to `does_k8s_exist`.

**Comment by to-bar** [comment]
After solving this issue, the fix should be backported to v0.8 branch.

**Comment by przemyslavic** [comment]
✔️ Tested. It works fine now - Prometheus is able to scrape metrics from managed k8s nodes.
✔️ The backport to v0.8.x was also tested here.
